templates/common/_base/units/kubens.service.yaml

```diff
@@ -0,0 +1,29 @@
+# Autogenerated by hack/update-templates.sh; do not edit
+name: kubens.service
+enabled: false
+contents: |
+  [Unit]
+  Description=Manages a mount namespace for kubernetes-specific mounts
+  
+  [Service]
+  Type=oneshot
+  RemainAfterExit=yes
+  RuntimeDirectory=kubens
+  Environment=RUNTIME_DIRECTORY=%t/kubens
+  Environment=BIND_POINT=%t/kubens/mnt
+  Environment=ENVFILE=%t/kubens/env
+  
+  # Set up the runtime directory as an unbindable mountpoint
+  ExecStartPre=bash -c "findmnt ${RUNTIME_DIRECTORY} || mount --make-unbindable --bind ${RUNTIME_DIRECTORY} ${RUNTIME_DIRECTORY}"
+  # Ensure the bind point exists
+  ExecStartPre=touch ${BIND_POINT}
+  # Use 'unshare' to create the new mountpoint, then 'mount --make-rshared' so it cascades internally
+  ExecStart=unshare --mount=${BIND_POINT} --propagation slave mount --make-rshared /
+  # Finally, set an env pointer for ease-of-use
+  ExecStartPost=awk 'BEGIN {print "KUBENSMNT=${BIND_POINT}" > "${ENVFILE}"}'
```

Comment:
Why the use of `awk` here?

Reply:
It's roughly the same as `bash -c 'echo KUBENSMNT=${BIND_POINT} > ${ENVFILE}'` but awk is lighter-weight than bash.

So more of a code-golf "what's better than bash for writing a file from systemd" than anything else.

Reply:
Is awk in the default install of RH?

Reply:
I would be shocked to find a linux system that doesn't have awk as part of the default toolset!

Reply:
Color me shocked!  At least in RHEL7, it turns out gawk is NOT one of the core packages.  I'll reimplement in bash shortly.

Reply:
Fixed.